### ISSUE TITLE
In vim mode, using @: to repeat last command doesn't work, breaks block cursor

### ISSUE THREAD

**Issue by dbarnett** [codemirror/CodeMirror#3219]
When I try repeating the last command with `@:` (to execute the command in the `:` register), it doesn't seem to execute anything, inserts a literal colon, switches to the thin cursor (but still acts like it's in normal mode).

**Comment by mightyguava** [comment]
Repeat works for me with `@:`. Can you provide reproducible steps + browser info? Did you get any errors?

**Comment by dbarnett** [comment]
Oh yeah, sorry, there is an error:
```
Uncaught TypeError: Cannot read property 'changes' of undefined
executeMacroRegister @ vim.js:4775
actions.replayMacro @ vim.js:2139
commandDispatcher.processAction @ vim.js:1179
commandDispatcher.processCommand @ vim.js:1099
(anonymous function) @ vim.js:826
runInOp @ codemirror.js:3120
CodeMirror.operation @ codemirror.js:5202
vimApi.findKey @ vim.js:820
doHandleBinding @ codemirror.js:3990
(anonymous function) @ codemirror.js:4061
CodeMirror.lookupKey @ codemirror.js:5744
lookupKeyForEditor @ codemirror.js:4004
dispatchKey @ codemirror.js:4020
handleCharBinding @ codemirror.js:4060
onKeyPress @ codemirror.js:4113
(anonymous function) @ codemirror.js:3130
```

Happens in Chrome 42 on Linux if I visit http://codemirror.net/demo/vim.html and do
```vim
:s/char/foo/
@:
```

**Comment by mightyguava** [comment]
Ah, there isn't actually support for repeating Ex commands with macros, but I've added support for `@:` since it was simple enough.

**Comment by dbarnett** [comment]
Fixed in 0185ceeae1.
